File: examples/15-java-client/src/main/java/io/orionmesh/examples/Consumer.java
package io.orionmesh.examples;

import io.orionmesh.client.OrionClient;
import io.orionmesh.client.Queue;

import java.util.HashMap;
import java.util.Map;

public class Consumer {
    public static void main(String[] args) throws Exception {
        String queue = System.getenv().getOrDefault("ORION_QUEUE_NAME", "events");
        String group = System.getenv().getOrDefault("ORION_QUEUE_GROUP", "java-consumer-workers");

        try (OrionClient c = new OrionClient()) {
            Queue q = c.queue(queue);
            Map<String, Integer> counts = new HashMap<>();
            for (Map<String, Object> row : q.sub(group, 0 /* forever */)) {
                String msg = String.valueOf(row.get("msg"));
                String basename = msg.contains("-") ? msg.split("-", 2)[0] : msg;
                counts.merge(basename, 1, Integer::sum);
                System.out.printf("got: %s  counts=%s%n", row, counts);
            }
        }
    }
}

File: examples/14-python-client/consumer.py
"""Consumer Service for the 14-python-client walkthrough.

Run via:
  target/debug/orion apply -f consumer-service.yaml
  target/debug/orion dispatch Service py-consumer
"""
from __future__ import annotations

import os
from collections import Counter

from orion_mesh import Client


def main():
    queue = os.environ.get("ORION_QUEUE_NAME", "events")
    group = os.environ.get("ORION_QUEUE_GROUP", "py-consumer-workers")

    c = Client()
    q = c.queue(queue)

    counts: Counter = Counter()
    for row in q.sub(group=group):
        msg = row.get("msg", "?")
        basename = msg.split("-", 1)[0] if isinstance(msg, str) else "?"
        counts[basename] += 1
        print(f"got: {row}  count={dict(counts.most_common(3))}", flush=True)


if __name__ == "__main__":
    main()
